runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -133,6 +134,35 @@ public class MqttKafkaSessionFactory implements MqttKafkaStreamFactory
     private static final int MQTT_KAFKA_CAPABILITIES = RETAIN_AVAILABLE_MASK | WILDCARD_AVAILABLE_MASK |
         SUBSCRIPTION_IDS_AVAILABLE_MASK;
     public static final String GROUPID_SESSION_SUFFIX = "session";
+    public static final Int2IntHashMap MQTT_REASON_CODES;
+    public static final Int2ObjectHashMap<String16FW> MQTT_REASONS;
+    public static final int GROUP_AUTH_FAILED_ERROR_CODE = 30;
+    public static final int INVALID_DESCRIBE_CONFIG_ERROR_CODE = 35;
+    public static final int INVALID_SESSION_TIMEOUT_ERROR_CODE = 26;
+    public static final byte MQTT_NOT_AUTHORIZED = (byte) 0x87;
+    public static final byte MQTT_IMPLEMENTATION_SPECIFIC_ERROR = (byte) 0x83;
+    public static final String MQTT_INVALID_SESSION_TIMEOUT_REASON = "Invalid session expiry interval";
+    private static final String16FW EMPTY_STRING = new String16FW("");
+
+    static
+    {
+        final Int2IntHashMap reasonCodes = new Int2IntHashMap(-1);
```

Comment:
Please us a constant for default error code returned when there is no entry in the map for the requested key.

i.e. `-1` should be a constant representing the default mqtt error code we want to send, then the `& 0xff` below likely becomes unnecessary.

Reply:
& 0xff is still necessary, as we defined the error codes as hex, and they're overflowing and become negative values. E.g: `byte MQTT_NOT_AUTHORIZED = (byte) 0x87;`

Reply:
`0xff` is an `int`
`MQTT_NOT_AUTHORIZED && 0xff` is `byte && int` which gives `int`
`MqttResetEx.reasonCode` takes an `int` and does the `uint8` range checking in the generated method body.
```
    public Builder reasonCode(int value) {
      if (value < 0) {
        throw new IllegalArgumentException(String.format("Value %d too low for field \"reasonCode\"", value));
      }
      if (value > 0XFF) {
        throw new IllegalArgumentException(String.format("Value %d too high for field \"reasonCode\"", value));
      }
```

So we need an `int`, but we're storing these constants as `byte`, causing the wrapping-to-become-negative problem and forcing the `& 0xff`.

Please change the type of the constants from `byte` to `int` and remove the `& 0xff` as it is then no longer needed.

Reply:
> Please us a constant for default error code returned when there is no entry in the map for the requested key.

I think you might have missed this feedback. The `missingValue` passed to the constructor is the value returned when the key is not present in the map. So if we get an error code other than the explicit ones listed, then we'll get that `missingValue` back from the map.

Currently we are masking that to `0xff` so we'd return a default mqtt error code of `0xff`.

Please define a constant for the default mqtt error code we want to return and pass it to the `Int2IntHashMap` constructor instead of `-1`.

If this default mqtt error code matches some of the explicit mappings below, such as `MQTT_IMPLEMENTATION_SPECIFIC_ERROR`, then they do not also need to be listed explicitly.